### ISSUE TITLE
double chest pet corrupts data on storage disk of refined storage

### ISSUE THREAD

**Issue by Jannik44** [capnkirok/Inventory-Pets#561]
Hello, i put a double chest pet into the me storage system, as soon as the pet conatains more than  item the disk stops working and i get this error message:

![image](https://github.com/capnkirok/Inventory-Pets/assets/76906033/140ef134-7f02-4076-ab03-73ffa4c05f64)

**Comment by capnkirok** [comment]
Yeah, it's a known issue with my chest pets. They don't initialize until you first open them by player hand. Definitely problems interacting with other mods that mess with inventories. I'm going to keep this here as I'm currently debugging.

**Comment by Jannik44** [comment]
any updates? :)

**Comment by capnkirok** [comment]
Still looking into it. Trying to get a fix in January.

**Comment by xlNovaKanelx** [comment]
any update on this its april and still happening?

**Comment by K3wl1** [comment]
came across this problem too :(

**Comment by Jannik44** [comment]
> came across this problem too :(

if it isnt fixed by now i think we can see this mod as abandoned, would remove it from modpacks

**Comment by capnkirok** [comment]
Not abandoned. Just haven't been able to make a fix. One fix that I haven't tried yet would be to just prevent the chests from working with Refined Storage at all. It would prevent the crash.

**Comment by Jannik44** [comment]
thats good, glad to see activity :)

**Comment by capnkirok** [comment]
Ok, I can confirm that the missing hover name is the problem. I don't know what MC is doing on item creation to cause this to null out when a non-player opens the chest, so I've created a hack and will release a patch this weekend, the solution is:
- If Refined Storage is installed and a Chest Pet (any kind) enters a player inventory that does not have a Custom Name, the pet will automatically be given a Custom Name based on the default item name (e.g., "Chest Pet" if using en_us). Tested and working.

If you want to get around this before the patch, folks just need to give their pet a name (just hit the N key when you've got the item selected).
